java/client/src/main/java/io/vitess/client/VTGateConnection.java

```diff
@@ -118,14 +119,11 @@ public VTGateConnection(RpcClient client, Query.ExecuteOptions options) {
      */
     public VTGateConnection(RpcClient client, String targetString, Query.ExecuteOptions options) {
         this.client = checkNotNull(client);
-        this.session = Session.newBuilder()
-                .setTargetString(null == targetString ? "" : targetString)
-                .setOptions(null == options ? Query.ExecuteOptions.newBuilder().build() : options)
-                .build();
+        this.globalSession = new VTSession(targetString, options);
```

Comment:
Why do we keep this session, it doesn't appear to be used anywhere. Everything is calling the *WithSession functions?